src/main/java/com/github/virtualgamers/hrtg/rest/controller/core/UserController.java

```diff
@@ -0,0 +1,37 @@
+package com.github.virtualgamers.hrtg.rest.controller.core;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import com.github.virtualgamers.hrtg.core.model.User;
+import com.github.virtualgamers.hrtg.core.service.UserService;
+
+/**
+ * API for the {@link UserService}
+ * 
+ * @author CCThomas
+ *
+ */
+@RestController
+@RequestMapping("/user")
+public class UserController {
+
+    Logger logger = LoggerFactory.getLogger(UserController.class);
+
+    @Autowired
+    UserService userService;
+
+    @GetMapping(path = "/getUser/{partyId}/{username}", produces = "application/json")
```

Comment:
it seems odd to put the partyId between the user resource and the usn? maybe those should be query params?
@GET /users?partyId={party}&username={usn}

Reply:
I have followed the pattern that if it is required, you put it in the url path, where optional parameters are query params. Party Id and Username make up the Primary Key for a user. So you could also look at is as "getUser/123" if "123" was the id. Looking at this though, I think I should change "getUser" to "user"

Reply:
that's true. is it more likely that we'll have access to the partyId and usn over the user id/pk?
also I think plural /users/ is more common

Reply:
Sorry. Could have been more clear. "is it more likely that we'll have access to the partyId and usn over the user id/pk" - Party Id and Username are the Primary Keys. The User Entity has 2 variables/columns, forming a unique combination, which equals the Primary Key.

In the reactor you can see the [here](https://github.com/virtualgamers/hidden-role-to-go-reactor/blob/issue-2/hidden-role-to-go-core/src/main/java/com/github/virtualgamers/hrtg/core/model/User.java#L68) where I have the variable for the User's id. Which is an Embeddable Class I define [above](https://github.com/virtualgamers/hidden-role-to-go-reactor/blob/issue-2/hidden-role-to-go-core/src/main/java/com/github/virtualgamers/hrtg/core/model/User.java#L29)

Here is a brief tutorial: https://www.baeldung.com/jpa-composite-primary-keys